tests/examples/sudoku/model.py

```diff
@@ -0,0 +1,57 @@
+# Based on example at page 112 in book:
+#       N. Sudermann-Merx: Einführung in Optimierungsmodelle, Springer Nature, 2023
+
+import os
+from pathlib import Path
+
+import polars as pl
+
+import pyoframe as pf
+from pyoframe import sum
+
+
+def main(input_dir, directory, use_var_names=True, **kwargs):
+    init_values = pl.read_csv(input_dir / "initial_numbers.csv")
+    init_values = init_values.with_columns(init_values=1)
+
+    digits = pl.int_range(1, 10, eager=True)
+
+    cube9x9x9 = (
+        pl.DataFrame({"row": digits})
+        .join(pl.DataFrame({"column": digits}), how="cross")
+        .join(pl.DataFrame({"digit": digits}), how="cross")
+    )
+    cube9x9x9 = cube9x9x9.with_columns(
+        box=((pl.col("row") - 1) // 3 + 1) * 10 + ((pl.col("column") - 1) // 3 + 1)
+    )
```

Comment:
Instead of adding the `box` column to this dataframe (which means now your variable has an extra unecessary column), I would create a new dataframe with columns `row,column,box`. This dataframe can then be used as a map with [`.map()`](https://bravos-power.github.io/pyoframe/reference/#pyoframe.Expression.map)